web/courses.go

```diff
@@ -98,7 +98,7 @@ func (s *AutograderService) updateEnrollment(ctx context.Context, sc scm.SCM, cu
 	case pb.Enrollment_NONE:
 		return s.rejectEnrollment(ctx, sc, enrollment)
 
-	case pb.Enrollment_STUDENT:
+	case pb.Enrollment_STUDENT, pb.Enrollment_PENDING:
```

Comment:
Are you sure this is correct behavior? If we receive a request from the frontend (or other client) with status pending, then we should enroll student? I know that we set the student to pending when we do `CreateEnrollment`, and that we read back the enrollment on Line 88. 

I guess my question is: will it happen that the frontend generates `PENDING` enrollment request, when we want to update it to student? I could imagine that we should instead just ignore such requests, like this:

```go
	case pb.Enrollment_PENDING:
		return nil
```

Reply:
I've revised and pushed code that hopefully makes the logic more sensible and a bit more decoupled.